### PR TITLE
Version check

### PR DESCRIPTION
**Pull request by j4qfrost**
This includes some clippy stuff and the eglCurrentDisplay patch as well.

### CODE REVIEW

File: src/renderer/fonts/caching_shaper.rs [hunk 1]
```diff
@@ -15,7 +15,7 @@ cfg_if! {
     if #[cfg(target_os = "windows")] {
         pub const SYSTEM_DEFAULT_FONT: &str = "Consolas";
     } else if #[cfg(target_os = "linux")] {
-        pub const SYSTEM_DEFAULT_FONT: &str = "Ubuntu";
+        pub const SYSTEM_DEFAULT_FONT: &str = "Mono";
```
Comment (j4qfrost) [inline src/renderer/fonts/caching_shaper.rs:18]: I didn't mean for these changes to make it in. I was working on something else.

File: src/bridge/mod.rs [hunk 2]
```diff
@@ -182,13 +183,14 @@ async fn start_neovim_runtime(
         close_watcher_running.store(false, Ordering::Relaxed);
     });
 
-    if let Ok(Value::Integer(correct_version)) = nvim.eval("has(\"nvim-0.4\")").await {
-        if correct_version.as_i64() != Some(1) {
-            error!("Neovide requires version 0.4 or higher");
+    if let Ok(output) = nvim.command_output("version").await {
```
Comment (Kethku) [inline src/bridge/mod.rs:186]: I'm slightly suspicious of this change. neovim defines options for all of the major versions, so any version greater than 0.4 should cause "has("nvim-0.4")" to return true.

Is this a fix for something I missed? If not, I think I should track down the documentation that indicates this behavior and add it as a comment.

File: src/bridge/mod.rs [hunk 3]
```diff
@@ -182,13 +183,14 @@ async fn start_neovim_runtime(
...
+        let re = Regex::new(r"NVIM v0.[4-9]\d*.\d+").unwrap();
+        if !re.is_match(&output) {
+            error!("Neovide requires nvim version 0.4 or higher. Download the latest version here https://github.com/neovim/neovim/wiki/Installing-Neovim");
             std::process::exit(0);
         }
     } else {
-        error!("Neovide requires version 0.4 or higher");
+        error!("Neovide requires nvim version 0.4 or higher. Download the latest version here https://github.com/neovim/neovim/wiki/Installing-Neovim");
```
Comment (Kethku) [inline src/bridge/mod.rs:193]: Good call giving further instructions

File: src/renderer/cursor_renderer/mod.rs [hunk 4]
```diff
@@ -20,8 +20,9 @@ const DEFAULT_CELL_PERCENTAGE: f32 = 1.0 / 8.0;
 
 const STANDARD_CORNERS: &[(f32, f32); 4] = &[(-0.5, -0.5), (0.5, -0.5), (0.5, 0.5), (-0.5, 0.5)];
 
-#[setting_prefix = "cursor"]
-#[derive(Clone, SettingGroup)]
+#[derive(SettingGroup)]
+#[setting_prefix = "keyboard"]
```
Comment (Kethku) [inline src/renderer/cursor_renderer/mod.rs:24]: correct me if I'm wrong, but I believe this represents a breaking change

File: src/renderer/mod.rs [hunk 5]
```diff
@@ -18,8 +18,9 @@ use crate::editor::{Colors, DrawCommand, Style, WindowDrawCommand};
 use crate::settings::*;
 use cursor_renderer::CursorRenderer;
 
-#[setting_prefix = "window"]
-#[derive(Clone, SettingGroup)]
+#[derive(SettingGroup)]
+#[setting_prefix = "keyboard"]
```
Comment (Kethku) [inline src/renderer/mod.rs:22]: This seems definitely incorrect. Maybe a merge error?

File: src/renderer/cursor_renderer/mod.rs (hunk 4 above)
Comment (j4qfrost) [reply src/renderer/cursor_renderer/mod.rs:24]: It behaves the same when I run it. Clippy was complaining about it and saying that the previous line would be counted as an error in the future.

File: src/window/keyboard/settings.rs [hunk 6]
```diff
@@ -1,8 +1,9 @@
 use super::KeyboardLayout;
 use crate::settings::FromValue;
 
+#[derive(SettingGroup)]
```
Comment (Kethku) [inline src/window/keyboard/settings.rs:4]: Interesting to split these up. Was that a clippy suggestion?

File: src/renderer/mod.rs (hunk 5 above)
Comment (j4qfrost) [reply src/renderer/mod.rs:22]: Yes. I think so lol.

**Comment by Kethku** [comment]
Good stuff. I appreciate the maintenance work

File: src/renderer/mod.rs (hunk 5 above)
Comment (j4qfrost) [reply src/renderer/mod.rs:22]: I think I was just brainlessly copy and pasting.

**Comment by j4qfrost** [comment]
Thanks, I'll make the fixes and push.

File: src/window/keyboard/settings.rs (hunk 6 above)
Comment (j4qfrost) [reply src/window/keyboard/settings.rs:4]: It was more about the order of the derivation. It wanted SettingsGroup to be resolved before the prefix was used.

File: src/bridge/mod.rs (hunk 2 above)
Comment (j4qfrost) [reply src/bridge/mod.rs:186]: While trying to build a snap, I ran into an issue when using the latest neovim version from git. The repo was marked with version 0.5.0, which wasn't caught by has.

File: src/renderer/cursor_renderer/mod.rs (hunk 4 above)
Comment (Kethku) [reply src/renderer/cursor_renderer/mod.rs]: to be specific, setting_prefix is changed from cursor to keyboard here. Thats what I was pointing out.

File: src/renderer/mod.rs (hunk 5 above)
Comment (Kethku) [reply src/renderer/mod.rs]: Ok cool. Just making sure to capture, window was changed to keyboard here.

File: src/window/window_wrapper/renderer.rs [hunk 7]
```diff
@@ -51,6 +51,9 @@ impl SkiaRenderer {
         gl::load_with(|s| windowed_context.get_proc_address(&s));
 
         let interface = skia_safe::gpu::gl::Interface::new_load_with(|name| {
+            if name == "eglGetCurrentDisplay" {
```
Comment (j4qfrost) [inline src/window/window_wrapper/renderer.rs:54]: I think the PR should be ready, though I this fix should be studied a bit more.

**Comment by j4qfrost** [comment]
@Kethku I'll continue my research with eglGetCurrentDisplay. I may need to make an upstream commit. The snap should be ready to merge too. I would need a review for the new github workflow though. If you would like to have admin access to the snapcraft package, just send me your email.

**Pull request merged**